### DISCUSSION TITLE
How can I add a UpdaterDelegate to an updater controller

### DISCUSSION THREAD

**Discussion by alaomoues** [sparkle-project/Sparkle#2431] [Q&A]
I’m trying add a delegate to a SwiftUI updater controller for monitoring what process that updater is. But after I attached it, it did not work correctly, it cannot call the protocol normally (nothing happened after the process that should call method). Idk why it like this, I think my code is absolutely correct. 
This is my code:

```swift
let updaterController = SPUStandardUpdaterController(
    startingUpdater: true,
    updaterDelegate: UpdaterDelegate(),
    userDriverDelegate: nil
)

class UpdaterDelegate: NSObject, SPUUpdaterDelegate {
    // delegate methods declared here
}

```

**Comment by zorgiepoo** [comment]
Ensure your updaterController and your updater delegate are kept alive by your app. Looking at your code, it is clear you are not ensuring the latter. It's standard practice for delegates to be weakly referenced.
